### ISSUE TITLE
Overlapping routes

### ISSUE THREAD

**Issue by jeff-h** [httpswift/swifter#294]
Similar to the example code:

```swift
server["/desktop/:path"] = shareFilesFromDirectory("/Users/me/Desktop")
```

I would like to do something like this (but it doesn't work, even reversing the order of the two statements):

```swift
server["/:path"] = shareFilesFromDirectory("/Users/me/Sites/test")
server["/vendor/:path"] = shareFilesFromDirectory("/Users/me/Sites/vendor")
```

This would enable the HTML inside the `test` directory to assume it's in the web root, and would also allow it to reference vendor libs (eg jQuery) from `/vendor/jquery.js`.

Unfortunately Swifter doesn't allow this, as it seems the first handler whose regex matches the path will be used, even if it returns a `404`.

Symlinks might be a solution except I have other factors which preclude them.

So my question is: is it possible to craft a symlink to match all, but to preclude a specific path? eg something like:

```swift
// match anything, so long as it doesn't begin with `/vendor` 
server["!{/vendor}/:path"] = shareFilesFromDirectory("/Users/me/Sites/test")

server["/vendor/:path"] = shareFilesFromDirectory("/Users/me/Sites/vendor")
```

My regex is awful at best, so hoping someone might be able to help.

**Comment by fabianthdev** [comment]
+1 For this question. I have exactly the same problem. Did you find a way to solve this?

**Comment by jeff-h** [comment]
I actually decided to use https://github.com/BiAtoms/Http.swift which supports overlapping paths.

**Comment by kimaldis** [comment]
Oddly enough, I'm trying to do exactly the same thing. No thoughts yet?

**Comment by Vkt0r** [comment]
The issue with the overlapping paths in the routes was solved in #382. Feel free to open in case it wasn't solved for you.
